app/src/main/kotlin/org/koitharu/kotatsu/local/data/input/LocalMangaDirInput.kt

```diff
@@ -91,16 +91,12 @@ class LocalMangaDirInput(root: File) : LocalMangaInput(root) {
 	override suspend fun getPages(chapter: MangaChapter): List<MangaPage> = runInterruptible(Dispatchers.IO) {
 		val file = chapter.url.toUri().toFile()
 		if (file.isDirectory) {
-			file.listFilesRecursive(ImageFileFilter())
+			file.walkCompat()
+				.filter { hasImageExtension(it) }
```

Comment:
Replacing a normal class within an anonymous class is a bit controversial solution...

Reply:
It's actually a [top-level function](https://kotlinlang.org/docs/functions.html#function-scope), which gets compiled to a static method by the Kotlin compiler.

Reply:
Yes, I mean `filter` function uses lambda that compiles into anonymous class

Reply:
But ok, It is not an issue in general